pol-core/bscript/compiler/astbuilder/ValueBuilder.cpp

```diff
@@ -23,10 +27,97 @@ std::unique_ptr<FloatValue> ValueBuilder::float_value(
   return std::make_unique<FloatValue>( loc, value );
 }
 
+std::unique_ptr<StringValue> ValueBuilder::string_value(
+    antlr4::tree::TerminalNode* string_literal )
+{
+  auto loc = location_for( *string_literal );
+  return std::make_unique<StringValue>( loc, unquote( string_literal ) );
+}
+
+std::string ValueBuilder::unquote( antlr4::tree::TerminalNode* string_literal )
+{
+  std::string input = string_literal->getSymbol()->getText();
+  const char* s = input.c_str();
+  if ( *s != '\"' )
+    location_for( *string_literal ).internal_error( "string does not begin with a quote?" );
+
+  const char* end = s + 1;
+  std::string lit;
```

Comment:
This function could use some refactoring later and I'm aware it's the same code as the old compiler. Should we at least reserve the string size?
```suggestion
  std::string lit;
  lit.reserve(input.length());
```